tracing-subscriber/src/layer/layered.rs

```diff
@@ -91,6 +91,11 @@ where
     L: Layer<S>,
     S: Subscriber,
 {
+    fn on_register_dispatch(&self, subscriber: &Dispatch) {
+        self.layer.on_register_dispatch(subscriber);
```

Comment:
I think we probably want to switch the order here, so that the actual subscriber gets the chance to register first.